### PR TITLE
[dashboards] Implement formulas and functions support for query table widgets

### PR DESCRIPTION
**Pull request by DrkSephy**
This pull request implements support for formulas and functions in the `Query Table` dashboard widget.

### CODE REVIEW

**Review by matt-miller-ddog — DISMISSED** [review]

**Review by cswatt — APPROVED** [review]

**Review by matt-miller-ddog — DISMISSED** [review]

File: datadog/resource_datadog_dashboard.go [hunk 1]
```diff
@@ -3818,6 +3821,29 @@ func buildDatadogQueryTableRequests(terraformRequests *[]interface{}) *[]datadog
...
+				if w, ok := query["event_query"].([]interface{}); ok && len(w) > 0 {
+					queries[i] = buildDatadogEventQuery(w[0].(map[string]interface{}))
+				} else if w, ok := query["metric_query"].([]interface{}); ok && len(w) > 0 {
+					queries[i] = buildDatadogMetricQuery(w[0].(map[string]interface{}))
+				} else if w, ok := query["process_query"].([]interface{}); ok && len(w) > 0 {
+					queries[i] = buildDatadogFormulaAndFunctionProcessQuery(w[0].(map[string]interface{}))
+				}
+			}
+			datadogQueryTableRequest.SetQueries(queries)
+			// Query Table requests for formulas and functions always has a response format of "scalar"
```
Comment (jirikuncar) [inline datadog/resource_datadog_dashboard.go:3837]:
```suggestion
			// Query Table request for formulas and functions always has a response format of "scalar"
```
```suggestion
			// Query Table requests for formulas and functions always have a response format of "scalar"
```

**Review by jirikuncar — APPROVED** [review]
It looks good to me.

Can you check the comment above?

**Review by jirikuncar — APPROVED** [review]

**Pull request merged**